library/src-bootstrapped/scala/quoted/util/package.scala

```diff
@@ -0,0 +1,17 @@
+package scala.quoted
+
+import scala.internal.quoted.showName
+
+package object util {
+
+  /** Genrate the code `'{ val `<name>` = $expr; ${ body('<name>) } }` with the given name */
+  def let[T: Type, U: Type](name: String)(expr: Expr[T])(body: Expr[T] => Expr[U])(given QuoteContext): Expr[U] = '{
+     @showName(${Expr(name)}) val x = $expr
+     ${ body('x) }
+  }
```

Comment:
What about creating a version of `let` in tasty reflect, and use it to implement this `let` instead of using `showName`?

Reply:
It would only solve the problem for the let but not any other `val`/`def`/`var` in any arbitrary quoted code.